src/forms/CreateVaultForm.tsx

```diff
@@ -0,0 +1,62 @@
+import { z } from 'zod';
+
+import { IFormToken } from 'components/FormAssetSelector/FormAssetSelector';
+import { ZERO_ADDRESS } from 'config/constants';
+import {
+  getToken,
+  hasToken,
+  validNetworkId,
+  zAssetEnum,
+  TAssetEnum,
+} from 'config/networks';
+
+import { createForm } from './createForm';
+import { zEthAddress } from './formHelpers';
+
+const schema = z
+  .object({
+    asset: z.object({
+      name: zAssetEnum,
+      custom: zEthAddress.optional(),
+    }),
+  })
+  .strict();
+
+const getZodParser = (chainId?: number) => {
+  return schema
+    .refine(() => validNetworkId(chainId), 'Invalid network Id')
+    .refine(
+      ({ asset: { name } }) => name === 'OTHER' || hasToken(name),
+      'Unknown Token'
+    )
+    .refine(({ asset: { name, custom } }) => name !== 'OTHER' || custom, {
+      path: ['asset'],
+      message: 'Token address is required for OTHER',
+    })
+    .transform(({ asset: { name, custom } }) =>
```

Comment:
This is combersome and I have an idea how to improve. Currently if a transform throws an error it get hidden, which is sad :'(

Reply:
Improved with `zTransformCatch`